src/plugin/timezone/index.js

```diff
@@ -33,6 +33,28 @@ const getDateTimeFormat = (timezone, options = {}) => {
   return dtf
 }
 
+const localeStringifierCache = {}
+const getLocaleStringifier = (timezone) => {
+  const localeStringifier = localeStringifierCache[timezone]
+  if (localeStringifier) {
+    return localeStringifier
+  }
+
+  const newLocaleStringifier = new Intl.DateTimeFormat('en-US', {
+    timeZone: timezone,
+    day: 'numeric',
```

Comment:
should we leave these configs to default?

Reply:
I simply configured it with trial-and-error until it 1) worked and 2) produced the same results that the old code did while doing some 10000 iterations on random dates and benchmarking the performance. I lack expertise and context here to judge this any further.

Reply:
Thanks, if so, share the same logic with  getDateTimeFormat ?

Reply:
I of course tried reusing the `getDateTimeFormat()` as-is before trying anything else, but everything crashed and burned. Then I thought of reusing *some* of the same logic and caching with timezone, format options and options, but decided against that because it would just have made it more complex.

Reply:
I'm not 100% sure whether my message is easily understood, so I will try to be clearer.

It was meant to be a question. Would you prefer me to refactor this to share logic between the two caching functions? I do not think that it will make the code clearer or much shorter in this case, so I will sit on this until further notice.

Reply:
Hi. I'm not too familiar with Github so please forgive me if this is not the best place to post this.

I'm using dayjs for a website and I've been making some changes to it recently, specifically adding timezones to some functions. With the latest addition, I've noticed a drastic performance issue.

I came across this thread, but I realized this solution hasn't been applied to the production code I have in my node_modules/dayjs/plugin/timezone.js file (version 1.11.4). So I decided to give it a try on my own:

Simply adding the getLocaleStringifier function, as is, on top of the code and editing "const target" to use it, I noticed a huge **performance gain of about 10 times**. I use dayjs.tz() several times in a loop, in nested functions and within some ES6 array methods, so my code may not be very efficient. Regardless, this loop execution time went down from more than 5 seconds to less than 600 ms.

I hope this helps and this improvement reaches the live dayjs version soon. Find also attached two screenshots from before and after the change.

![image](https://user-images.githubusercontent.com/30042505/182682229-81c8d203-02fa-4177-8491-1d57b59e2a89.png)

![image](https://user-images.githubusercontent.com/30042505/182682269-5ba08823-7a35-45e8-891c-014da31e5dee.png)

Reply:
<img width="451" alt="image" src="https://user-images.githubusercontent.com/5386958/183840527-62334b17-26a8-4538-ad62-06ab38094b8d.png">
It seems this 2 function not generate same format;

In my case, getLocaleStringifier worked fine, use getDateTimeFormat make date not correct